src/Cache/InMemoryCache.ts

```diff
@@ -3,18 +3,21 @@ import { ICache } from './ICache';
 export class InMemoryCache implements ICache {
 	private cache = {};
 
-	public set(key: string, value: any): void {
+	public set(key: string, value: any): any {
 		this.cache[key] = value;
+		return value !== null && typeof value === 'object' ? Object.assign({}, value) : value;
```

Comment:
Another thing: We should build cache asynchronous from the begining becaouse when we will for example switch to IDB we would have to rewrite all usages...